sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetry.java

```diff
@@ -212,12 +212,13 @@ public boolean isClientMetricsEnabled() {
         return this.clientMetricsEnabled;
     }
 
-    public void init() {
-        loadAzureVmMetaData();
+    public Mono<?> init() {
+        Mono<?> startSendingClientTelemetryToServiceMono = this.isClientTelemetryEnabled()
+            ? sendClientTelemetry()
+            : Mono.empty();
 
-        if (this.isClientTelemetryEnabled()) {
```

Comment:
I think the original way might be better else it will blocking the new log message

Reply:
Never mind - you are right...

Reply:
Fixed

Reply:
yea, by theory `loadAzureVmMetaData` does not need to change into Mono, the log can be added after this method being called ~~

Reply:
No - not really, because the log must happen after the VM info has been loaded

Reply:
Otherwise the MachineId is not available. But the sendClientTelemetry should not be waited on but happen as a side-effect - doOnTerminate would do that. Before we could have started sending client telemetry before getting VM instance metadata - which would result in "wrong" metadata

Reply:
oooo I see, sorry missed the fact that loading VM data is actually an async action, always thought we block on it, thanks for the explanation